src/modalities/models/components/layer_norms.py

```diff
@@ -0,0 +1,92 @@
+from typing import Annotated
+
+import torch
+import torch.nn as nn
+from pydantic import BaseModel, Field
+from torch.nn import functional as F
+
+from modalities.config.lookup_enum import LookupEnum
+
+
+class LayerNormIF(nn.Module):
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        raise NotImplementedError
+
+    def __copy__(self):
+        raise NotImplementedError
+
+
+class RMSLayerNorm(LayerNormIF):
+    def __init__(self, ndim: int, epsilon: float = 1e-6):
+        """
+        Initialize the RMSNorm normalization layer.
+        Original paper: https://arxiv.org/pdf/1910.07467.pdf
+        Source code adopted from https://github.com/facebookresearch/llama/blob/a0a4da8b497c566403941ceec47c2512ecf9dd20/llama/model.py#L34C1-L77C36
+
+        Args:
+            ndim (int): The dimension of the input tensor.
+            eps (float, optional): A small value added to the denominator for numerical stability. Default is 1e-6.
+
+        Attributes:
+            eps (float): A small value added to the denominator for numerical stability.
+            weight (nn.Parameter): Learnable scaling parameter.
+
+        """
+        super().__init__()
+        self.epsilon = epsilon
+        self.weight = nn.Parameter(torch.ones(ndim))
+
+    def _norm(self, x: torch.Tensor) -> torch.Tensor:
+        return x * torch.rsqrt(x.pow(2).mean(-1, keepdim=True) + self.epsilon)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        output = self._norm(x.float()).type_as(x)
+        return output * self.weight
+
+    def __copy__(self):
+        copied_instance = RMSLayerNorm(self.weight.shape[0], self.epsilon)
+        return copied_instance
+
+
+class ZLayerNorm(LayerNormIF):
+    """LayerNorm but with an optional bias. PyTorch doesn't support simply bias=False"""
```

Comment:
A custom LayerNorm class to support `bias=False` seems to be a relict from [nanoGPT](https://github.com/karpathy/nanoGPT/blob/325be85d9be8c81b436728a420e85796c57dba7e/model.py#L19). It seems that PyTorch supports this natively (see [LayerNorm](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html)) since version 2.1 (see [here](https://github.com/pytorch/pytorch/pull/101683)). Can’t we just use PyTorch’s `LayerNorm` instead of our own `ZLayerNorm` class?